src/lib.rs

```diff
@@ -202,6 +231,63 @@ impl Client {
         }
     }
 
+    /// Sets the given keys.
+    ///
+    /// If `ttl` or `flags` are not specified, they will default to 0.  If the value is set
+    /// successfully, `()` is returned, otherwise [`Error`] is returned.
+    pub async fn set_multi<I, K, V>(
+        &mut self,
+        kv: I,
+        ttl: Option<i64>,
+        flags: Option<u32>,
+    ) -> Result<(), Error>
+    where
+        I: IntoIterator<Item = (K, V)>,
+        K: AsRef<[u8]>,
+        V: AsMemcachedValue,
+    {
+        let mut kv_iter = kv.into_iter().peekable();
+
+        if kv_iter.peek().is_none() {
+            return Ok(());
+        }
+
+        let mut num_results = 0;
+
+        for (key, value) in kv_iter {
+            let kr = key.as_ref();
+            let vr = value.as_bytes();
+
+            self.conn.write_all(b"set ").await?;
+            self.conn.write_all(kr).await?;
+
+            let flags = flags.unwrap_or(0).to_string();
+            self.conn.write_all(b" ").await?;
+            self.conn.write_all(flags.as_ref()).await?;
+
+            let ttl = ttl.unwrap_or(0).to_string();
+            self.conn.write_all(b" ").await?;
+            self.conn.write_all(ttl.as_ref()).await?;
+
+            let vlen = vr.len().to_string();
+            self.conn.write_all(b" ").await?;
+            self.conn.write_all(vlen.as_ref()).await?;
+            self.conn.write_all(b"\r\n").await?;
+
+            self.conn.write_all(vr.as_ref()).await?;
+            self.conn.write_all(b"\r\n").await?;
+
+            num_results += 1;
+        }
+        self.conn.flush().await?;
+
+        match self.get_write_multi_responses(num_results).await? {
```

Comment:
What is the desired API for `set_multi`? 

This seems to just return a single error even if we are successful for all of the other keys, but we do not know what keys were successful/errored?

Maybe we can re-think the api here to something that makes a bit more sense to the developer.

In my opinion we should either:

1. Lean hard into the `noreply` and just set and forget without even worrying about errors from the server, or maybe as you have just sending back one error to know if any failed.
2. Be more verbose and return an array of errors/failures per key that was sent?

Reply:
I've pushed up a couple of methods that lean in to option 2, but going with a HashMap approach since that will give us the most consistent information.  We could also include a noreply version for the option of faster execution times?